docs/rofl/features/appd.md

```diff
@@ -162,3 +162,43 @@ For example:
   `{"fail": {"code": 8, "module": "evm", "message": "reverted: aJAoLw=="}}`.
 
 [call result]: https://api.docs.oasis.io/rust/oasis_runtime_sdk/types/transaction/enum.CallResult.html
+
+## Instance Metadata
+
+Instance metadata allows apps to publish arbitrary key-value pairs that are
+included in the on-chain ROFL instance registration. This metadata is
+automatically namespaced with `net.oasis.app.` when published on-chain.
+
+### Get Metadata
+
+Retrieve all user-set metadata key-value pairs.
+
+**Endpoint:** `/rofl/v1/metadata` (`GET`)
+
+**Example response:**
+
+```json
+{
+  "key_fingerprint": "a54027bff15a8726",
+  "version": "1.0.0"
+}
+```
+
+### Set Metadata
+
+Set metadata key-value pairs. This replaces all existing app-provided metadata
+and will trigger a registration refresh if the metadata has changed.
+
+**Endpoint:** `/rofl/v1/metadata` (`POST`)
+
+**Example request:**
+
+```json
+{
+  "key_fingerprint": "a54027bff15a8726",
+  "version": "1.0.0"
+}
+```
+
+**Note:** Metadata is validated against runtime-configured limits for the
+number of pairs, key size, and value size.
```

Comment:
I would also add the "Remove Metadata" endpoint using the HTTP DELETE request. Similar to how the env variables can be set as empty string or not set at all.

Reply:
The POST metadata overrides existing metadata, so this can be done by POSTing an empty map. Not sure if we need to add a HTTP DELETE request endpoint to `rofl-appd`.  The client can expose a delete method (and just call POST `{}`) to achieve the same thing for the enduser.